### ISSUE TITLE
Import parsing issue when imports are not at the top of the file

### ISSUE THREAD

**Issue by BartGo** [bndr/pipreqs#36]
Windows 7, pipreqs 0.3.9, file to process (example/main.py):
```
import logging
logging.basicConfig()
from pydal import DAL, Field # some comment
from transitions import Machine # some comment

print "Hello"
```

Running:
```
> pipreqs --savepath req-pipreqs.txt --debug example

DEBUG: Found packages: set(['logging'])
DEBUG: Found imports:
DEBUG: Getting packages information from Local/PyPI
DEBUG: Writing 0 requirements:  to req-pipreqs.txt
INFO: Successfully saved requirements file in req-pipreqs.txt

```
Logging is not saved into req-pipreqs.txt which is fine because it's a standard library. Imports of pydal and transitions are not taken into account which is an error.

Removing the second line (or commenting out), so "logging.basicConfig()" causes pydal and transitions to be recognised correctly so it's a source parsing issue.
 
Seems to be related to the fix of #35 but also may be a slightly different scenario.

**Comment by bndr** [comment]
That is correct. PEP8 states that imports should be at the top of the file, so if you start calling functions and defining variables, everything after that is assumed to be code not imports.

**Comment by BartGo** [comment]
Thanks for your quick answer!

I see that you're referring to E402, it's very restrictive (and there are some discussions around it) but anyway the issue I noticed is present not only for the cases violating PEP8/E402, it's wider.

Example:
```
import bottle
try:
    import json
except ImportError:
    import simplejson as json

print "Imports done."
b = bottle.Bottle()
j = json.dumps([])
```
And:
```
> flake8 -v example/main.py
```
No issues found (flake: 2.5.1, pep8: 1.7.0, pyflakes: 1.0.0, mccabe: 0.3.1). Then, pipreqs does not recognise any of the modules (bottle, json and/or simplejson):

```
> pipreqs --debug --savepath req.txt example/main.py
DEBUG: Found packages: set([])
DEBUG: Found imports:
DEBUG: Getting packages information from Local/PyPI
DEBUG: Writing 0 requirements:  to req.txt
INFO: Successfully saved requirements file in req.txt
```

**Comment by bndr** [comment]
You do have a point, another thing I forgot is lazy loading imports e.g:

```
def somefunc():
    import some_module
```

So this should definitely be looked into.

Thanks again for reporting.

**Comment by BartGo** [comment]
No problem! I renamed the issue as the previous title was misleading.

**Comment by bndr** [comment]
This should be fixed in version 0.4.0
